specs/math.txt

```diff
@@ -0,0 +1,293 @@
+Run this with `cargo run -- -M -s specs/math.txt`.
+
+Inline math is done with a single dollar.
+
+```````````````````````````````` example
+The equivalence of energy and mass: $E = mc^2$.
+.
+<p>The equivalence of energy and mass: <span class="math math-inline">E = mc^2</span>.</p>
+````````````````````````````````
+
+Inline math can be used multiple times.
+
+```````````````````````````````` example
+$x$ and $y$ are integral variables.
+.
+<p><span class="math math-inline">x</span> and <span class="math math-inline">y</span> are integral variables.</p>
+````````````````````````````````
+
+Inline math can span multiple lines...
+
+```````````````````````````````` example
+$x = \frac{-b \pm \sqrt{b^2 - 4ac}}
+          {2a}$
+.
+<p><span class="math math-inline">x = \frac{-b \pm \sqrt{b^2 - 4ac}} {2a}</span></p>
+````````````````````````````````
+
+...but cannot have empty lines.
+
+```````````````````````````````` example
+$x = \frac{-b \pm \sqrt{b^2 - 4ac}}
+
+{2a}$
+.
+<p>$x = \frac{-b \pm \sqrt{b^2 - 4ac}}</p>
+<p>{2a}$</p>
+````````````````````````````````
+
+Display math is done with a double dollar.
+
+```````````````````````````````` example
+The equivalence of energy and mass: $$E = mc^2.$$
+.
+<p>The equivalence of energy and mass: <span class="math math-display">E = mc^2.</span></p>
+````````````````````````````````
+
+Display math can span multiple lines...
+
+```````````````````````````````` example
+$$x = \frac{-b \pm \sqrt{b^2 - 4ac}}
+           {2a}$$
+.
+<p><span class="math math-display">x = \frac{-b \pm \sqrt{b^2 - 4ac}} {2a}</span></p>
+````````````````````````````````
+
+...and even have empty lines.
+
+```````````````````````````````` example
+$$x = \frac{-b \pm \sqrt{b^2 - 4ac}}
+
+{2a}$$
+.
+<p><span class="math math-display">x = \frac{-b \pm \sqrt{b^2 - 4ac}}  {2a}</span></p>
+````````````````````````````````
+
+Multi-line math works as expected in containers.
+
+```````````````````````````````` example
+1. > 3-by-3 matrix $A = \begin{pmatrix}
+   >                    a & b & c \\
+   >                    d & e & f \\
+   >                    g & h & i
+   >                    \end{pmatrix}$ has a determinant
+   > $$\mathrm{det} A
+   >
+   > = a \mathrm{det} \begin{pmatrix} e & f \\ h & i \end{pmatrix}
+   > - b \mathrm{det} \begin{pmatrix} d & f \\ g & i \end{pmatrix}
+   > + c \mathrm{det} \begin{pmatrix} d & e \\ g & h \end{pmatrix}
+   >
+   > = a(ei - fh) - b(di - fg) + c(dh - eg).$$
+   >
+   > In the other words, it is the sum of elements in the top row and
+   > determinants for respective *minors* with alternating signs.
```

Comment:
(Hmm, should have been "the sum of elements in the top row *times* determinants for respective minors". Just in case.)